docs/_tutorials/cache_api.md

```diff
@@ -0,0 +1,87 @@
+---
+layout: tutorials
+permalink: /tutorials/cache-api/
+title: Cache API
+---
+
+Jekyll includes a caching API, which is used both internally as well as exposed
+for plugins, which can be used to cache the output of deterministic functions to
+speed up site generation. This cache will be persistent across builds, but
+cleared when Jekyll detects any changes to `_config.yml`.
+
+## Jekyll::Cache.new(name) → new_cache
```

Comment:
Can we introduce sections with intent and keep the code snippet for examples?

Reply:
Was trying to mimic RubyDoc.
I am open to suggestions.